### PR TITLE
scheduler(ut): call close method when finish profileMap in ut

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
- call close method when finish profileMap in ut
code: https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/profile/profile.go#L74
#### Which issue(s) this PR fixes:

Fixes # 

#### Special notes for your reviewer:
None 
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/profile/profile_test.go [hunk 1]
```diff
@@ -251,6 +251,14 @@ func TestNewMap(t *testing.T) {
 			ctx, cancel := context.WithCancel(ctx)
 			defer cancel()
 			m, err := NewMap(ctx, tc.cfgs, fakeRegistry, nilRecorderFactory)
```
Comment (sanposhiho) [inline pkg/scheduler/profile/profile_test.go:253]: looks like we should check `err` from NewMap?

Comment (sanposhiho) [reply pkg/scheduler/profile/profile_test.go:253]: Ah, `checkErr` below does that.
Then, let's move defer after `checkErr`?

Comment (sanposhiho) [reply pkg/scheduler/profile/profile_test.go:253]: My point is, when `err != nil`, it's likely `m` is nil and `defer` would panic.

Comment (googs1025) [reply pkg/scheduler/profile/profile_test.go:253]: I will revise it and add some test cases

Comment (googs1025) [reply pkg/scheduler/profile/profile_test.go:253]: if `defer` func, i add `if m != nil `

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

**Comment by googs1025** [comment]
kind ping :)

**Comment by sanposhiho** [comment]
It's already approved.

**Comment by sanposhiho** [comment]
and, we're under the code freeze.

**Pull request merged**